05-Interfaces/Lowercase_converter/mymodule_sysfs.c

```diff
@@ -0,0 +1,85 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+
+char arr[PAGE_SIZE];
+
+// routines
+static void lowercaseConvert(char *pbuf)
+{
+	int i = 0;
+	while(pbuf[i] != 0)
+	{
+		if(pbuf[i] >= 65 && pbuf[i] <= 96)
+		{
+			pbuf[i] = pbuf[i] - 'A' + 'a';
+		}
+		i++;	
+	}
+}
+
+static ssize_t param_show(struct class *class, struct class_attribute *attr, char *buf)
+{
+	pr_info("mymodule: read data");
+
+	strcpy(buf, arr);
+
+	return strlen(buf);
+}
+
+static ssize_t param_store(struct class *class, struct class_attribute *attr, const char *buf, size_t count)
+{	
+	pr_info("mymodule: write data");
+	
+	strcpy(arr, buf);
```

Comment:
Please add check if count is larger than arr size.
And handle the buf as a char array with the given size. Don't treat it as a string.